File: dart/EncapsulationExample.dart
// Encapsulation implementation in dart

// Parent class
class RandomGuyIdk {

    String _name = "Guy";
    int _age = 19;
    bool _dead = false;

    // Getter and Setter for Name
    String get name => _name;

    set name(String newName) {
        this._name = newName;
    }

    // Getter and Setter for Age
    int get age => _age;

    set age(int newAge) {
        this._age = newAge;
    }

    // Getter and Setter for Dead
    bool get dead => _dead;

    set dead(bool newDead) {
        this._dead = newDead;
    }
}

void main() {
    RandomGuyIdk obj = RandomGuyIdk();

    // This will not work (if main is in different file)
    /*
      * obj._name = "MewMew";
      * obj._age = 18;
      * obj._dead = false;
      */

    // Must be directly accessed through getters and setters
    obj.name = "MewMew";
    obj.age = 18;
    obj.dead = false;

    print("${obj.name} is ${obj.age} years old.");
    print("Is ${obj.name} dead? ${obj.dead}");
}

File: java/EncapsulationExample.java
// Encapsulation implementation in java

// Parent class
class RandomGuyIdk {

    private String name;
    private int age;
    private boolean dead;

    // Getter and Setter for Name
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    // Getter and Setter for Age
    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        if (age > 0) {
            this.age = age;
        } else {
            System.out.println("Invalid age. Age must be positive.");
        }
    }

    // Getter and Setter for Dead
    public boolean isDead() {
        return dead;
    }

    public void setDead(boolean dead) {
        this.dead = dead;
    }
}

public class EncapsulationExample {
    public static void main(String[] args) {
        RandomGuyIdk obj = new RandomGuyIdk();

        // This will not work
        /*
         * obj.name = "MewMew";
         * obj.age = 18;
         * obj.dead = false;
         */

        // Must be directly accessed through getters and setters
        obj.setName("MewMew");
        obj.setAge(18);
        obj.setDead(false);

        System.out.println(obj.getName() + " is " + obj.getAge() + " years old.");
        System.out.println("Is " + obj.getName() + " dead? " + obj.isDead());
    }
}